KEH-1204 | Fix Survey Completion Missing

TotalDwarf03 opened the pull request:
- publishingData was not defined previously.
- This caused the survey completion to not work.
- This is now fixed.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### What

Before the patch, when getting to the survey summary page, the completions information would not appear.

This was because the function was trying to use a `publishingData` without defining it, causing an error.

This has now been fixed by adding a `publishingData` parameter to the function.

### Testing

Have any new tests been added as part of this issue? If not, try to explain why test coverage is not needed here.

- [ ] Yes
- [x] No
Please write a brief description of why test coverage is not necessary here.
- [ ] Not as part of this ticket. (Could be done at a later point)

### Documentation

Has any new documentation been written as part of this issue? We should try to keep documentation up to date 
as new code is added, rather than leaving it for the future.

- [ ] Yes
- [x] No
Please write a brief description of why documentation is not necessary here.
- [ ] Not as part of this ticket. (Could be done at a later point)

### Related issues

KEH-1204 (Jira)

### How to review

- Create new project
- Go through survey
- Double check Survey Summary functionality

TotalDwarf03 committed:
fix: add publishingData to changeBtnURL function

- publishingData was not defined previously.
- This caused the survey completion to not work.
- This is now fixed.

delterr reviewed (approved):
Looks good!

sebtheo reviewed (commented):
Maybe I didn't document the bug well enough, but if you fill in all information (or just the first section) the Project Details will still say 'Partially Completed' and then each button click for 'Fill in missing details' will redirect you to `/survey/project` despite it being filled in.

mwirikia commented on static/formScripts.js:
```diff
@@ -258,7 +258,7 @@ function validateMultipleFields(data, fields) {
 function changeBtnURL(contactTechData, contactManagerData, projectData, projectDependenciesData, 
     sourceControlData, databaseData, languagesData, 
-    frameworksData, integrationsData, infrastructureData, 
+    frameworksData, integrationsData, infrastructureData, publishingData, 
```
does it also need to go into the survey.html part?

TotalDwarf03 committed:
fix: Section completion logic and button URL updates

- Added missing parameters when calling changeBtnURL
- Update complete logic in storeContactData. Wasn't working when checking technical contact.
- Update projectData validateMultipleFields keys to match local storage keys.

Note:
- This only fixes the current survey completion logic.
- This doesn't check most of the newer fields which have been added to the survey.
- Unsure if we should update this to add the new fields or if we should just leave it as is for now.

sebtheo commented on static/formScripts.js:
```diff
@@ -292,7 +295,7 @@ function changeBtnURL(contactTechData, contactManagerData, projectData, projectD
             data: projectData, 
             url: '/survey/project', 
-            validationFn: (data) => validateMultipleFields(data, ['project_name', 'project_short_name', 'project_description', 'doc_link'])
+            validationFn: (data) => validateMultipleFields(data, ['name', 'short_name', 'project_description', 'documentation_link'])
```
should this also include programme name and programme short name?

sebtheo replied:
i havent checked by why was project_name refactored to just name?

does it not make more sense for it to be called project_name? no biggie can still merge

TotalDwarf03 replied:
Addressed

sebtheo reviewed (approved):
Happy to approve, one minor

TotalDwarf03 committed:
fix: update validation to include programme info

TotalDwarf03 committed:
chore(formScripts): Add comments to validation checks.

- Added comments to indicate missing validation checks.
- Added comments to match validation checks with their respective sections.
